### PR TITLE
feat: ProphetModelDataset

### PR DESCRIPTION
**Pull request by galenseilis**
## Description
This PR was created to add Meta's (formerly Facebook's) Prophet model. It is a probabilistic modelling for time series forecasting.

## Development notes

The Prophet module provides functions for serializing and deserializing models to/from JSON. Beyond that the behaviour is intended to be the same as the `JSONDataset`. I have subclassed from `JSONDataset` and overrode the `_load` and `_save` methods to use Prophet's serialization during these methods.

I have tried to emulate the tests for `JSONDataset`, but with a code duplicate for a trained Prophet model instead of a test file.

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by merelcht** [comment]
Hi @galenseilis, thanks for the PR. I think this would be a great candidate to go in as an [experimental dataset ](https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/CONTRIBUTING.md#experimental-datasets). This relaxes the requirements for tests. How does that sound?

**Comment by galenseilis** [comment]
> Hi @galenseilis, thanks for the PR. I think this would be a great candidate to go in as an [experimental dataset ](https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/CONTRIBUTING.md#experimental-datasets). This relaxes the requirements for tests. How does that sound?

Sounds good to me!

**Comment by merelcht** [comment]
Great! @galenseilis are you okay moving it to the experimental directory yourself or do you want me to jump in and make the PR ready for review?

**Comment by galenseilis** [comment]
> Great! @galenseilis are you okay moving it to the experimental directory yourself or do you want me to jump in and make the PR ready for review?

Hi @merelcht , I would appreciate it if you would jump in to make the PR ready for review. My apologies for any inconvenience.

**Comment by merelcht** [comment]
@galenseilis no worries, happy to help!

**Comment by merelcht** [comment]
@galenseilis, I've made all changes required. I've updated the tests and docstrings. The rest are mostly docs build and linting changes. Let me know if all looks good and I'll ask the team to review 🙂

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -96,5 +96,5 @@ repos:
         name: "Bandit security check"
         language: system
         types: [file, python]
-        exclude: ^kedro-(telemetry|docker|datasets|airflow)/tests/|^kedro-(telemetry|docker|datasets|airflow)/features/
+        exclude: ^kedro-(telemetry|docker|datasets|airflow)/tests/|^kedro-(telemetry|docker|datasets|airflow)/features/|kedro-datasets/kedro_datasets_experimental/tests/
```
Comment (ElenaKhaustova) [inline .pre-commit-config.yaml:99]: Maybe we use `# nosec B108 ` for the target line instead?

**Review by ElenaKhaustova — APPROVED** [review]
Thank you @galenseilis for the contribution and @merelcht for polishing it!

I left one nit comment but the implementation LGTM.

I approve the PR but as far as I understand we're it is blocked by https://github.com/kedro-org/kedro-plugins/issues/850 now, so we need to resolve it first.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ankatiyar) [reply .pre-commit-config.yaml]: let me try to fix this

**Review by ankatiyar — APPROVED** [review]
Thanks @galenseilis, I took the liberty to add this to the release notes and fixed `bandit` related security stuff! Happy to get this in for the next release!

**Pull request merged**